src/uwtools/drivers/driver.py

```diff
@@ -32,39 +33,61 @@ class Assets(ABC):
     def __init__(
         self,
         config: Optional[Union[dict, Path]],
-        dry_run: bool = False,
-        batch: bool = False,
         cycle: Optional[datetime] = None,
         leadtime: Optional[timedelta] = None,
         key_path: Optional[list[str]] = None,
+        dry_run: bool = False,
     ) -> None:
         """
         A component driver.
 
         :param config: Path to config file (read stdin if missing or None).
-        :param dry_run: Run in dry-run mode?
-        :param batch: Run component via the batch system?
         :param cycle: The cycle.
         :param leadtime: The leadtime.
         :param key_path: Keys leading through the config to the driver's configuration block.
+        :param dry_run: Run in dry-run mode?
         """
-        dryrun(enable=dry_run)
         self._config = YAMLConfig(config=config)
-        self._batch = batch
-        has_leadtime = leadtime is not None
-        if has_leadtime and not cycle:
-            raise UWError("When leadtime is specified, cycle is required")
         self._config.dereference(
             context={
                 **({"cycle": cycle} if cycle else {}),
-                **({"leadtime": leadtime} if has_leadtime else {}),
+                **({"leadtime": leadtime} if leadtime is not None else {}),
                 **self._config.data,
             }
         )
-        key_path = key_path or []
-        for key in key_path:
+        for key in key_path or []:
             self._config = self._config[key]
         self._validate()
+        dryrun(enable=dry_run)
+
+    def __repr__(self) -> str:
+        cycle = self._cycle.strftime("%Y-%m-%dT%H:%M") if hasattr(self, "_cycle") else None
+        leadtime = None
+        if hasattr(self, "_leadtime"):
+            h, r = divmod(self._leadtime.total_seconds(), 3600)
+            m, s = divmod(r, 60)
+            leadtime = "%02d:%02d:%02d" % (h, m, s)
+        return " ".join(
+            filter(None, [str(self), cycle, leadtime, "in", self._driver_config["run_dir"]])
+        )
+
+    def __str__(self) -> str:
+        return self._driver_name
+
+    @property
+    def config(self) -> dict:
+        """
+        Return a copy of the driver config.
+        """
+        return deepcopy(self._driver_config)
```

Comment:
This will return `<block>` in e.g. for the UPP driver given UW YAML
```
upp:
    <block>
```

It returns a copy so that the receiver cannot use the returned value to interfere with the state of the driver object.